### PR TITLE
Improve vault deletion method

### PR DESCRIPTION
**Pull request by binsky08**
- delete credentials before vault is deleted from backend
- reduce requests from one per credential/file to one request containing all credential guids and all file ids that should be deleted

### CODE REVIEW

File: controller/vaultcontroller.php [hunk 1]
```diff
@@ -11,32 +11,35 @@
...
 
 	public function __construct($AppName,
-								IRequest $request,
-								$UserId,
-								VaultService $vaultService,
-								CredentialService $credentialService,
-								DeleteVaultRequestService $deleteVaultRequestService,
-								SettingsService $settings) {
+	                            IRequest $request,
+		$UserId,
```
[the remark this answers is not in the document]
Comment (animalillo) [reply controller/vaultcontroller.php:39]: but still, it should be aligned, it's more of a visual aid than anyghing, but idk, it feels wrong to see it out of place

File: controller/credentialcontroller.php [hunk 2]
```diff
@@ -37,12 +37,12 @@ class CredentialController extends ApiController {
 
 	public function __construct($AppName,
 	                            IRequest $request,
-	                            $userId,
-	                            CredentialService $credentialService,
-	                            ActivityService $activityService,
-	                            CredentialRevisionService $credentialRevisionService,
-	                            ShareService $sharingService,
-	                            SettingsService $settings
+		$userId,
```
[the remark this answers is not in the document]
Comment (animalillo) [reply controller/credentialcontroller.php:40]: but still, it should be aligned, it's more of a visual aid than anyghing, but idk, it feels wrong to see it out of place

File: controller/filecontroller.php [hunk 3]
```diff
@@ -65,22 +67,39 @@ public function deleteFile($file_id) {
...
 
-	public function updateFile($file_id, $file_data, $filename){
-		try{
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 */
+	public function deleteFiles($file_ids) {
+		if ($file_ids != null && !empty($file_ids)) {
+			foreach (json_decode($file_ids) as $file_id) {
```
[the remark this answers is not in the document]
Comment (binsky08) [reply controller/filecontroller.php]: fixed

File: controller/vaultcontroller.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (binsky08) [reply controller/vaultcontroller.php]: fixed with 4152f19c426236797ba046d5f706282b0c0419c8

File: controller/credentialcontroller.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (binsky08) [reply controller/credentialcontroller.php]: fixed with 4152f19c426236797ba046d5f706282b0c0419c8

**Pull request merged**